### PR TITLE
Organize blog posts into folders

### PR DESCRIPTION
**Pull request by ryan-blunden**
The main code to be reviewed is the change to [gatsby-node.js](https://github.com/sourcegraph/about/pull/801/files#diff-c85ccc351c0c202d7efe6d26344b01e3)

**NOTE**: This change **does not** affect the URLs of any of our blog posts because that is controlled by the `slug` field which every blog post file has.

@nicksnyder I've quickly added more instructions for how to add a blog post to `README.md`, although I will revise and move to the handbook in a future PR.

### CODE REVIEW

**Comment by christinaforney** [comment]
Love this change!

Just curious how we're going to handle redirects for people who have bookmarked these pages and google page ranks? I imagine @aileenrose is better informed here than me to offer suggestions/requirements.

File: README.md [hunk 1]
```diff
@@ -18,25 +18,84 @@ Then to serve the site, you have two options:
...
+## Publishing a blog post
+
+We want and encourage everyone at Sourcegraph to contribute to the blog and publishing a blog post is as simple as merging an approved pull request with your blog content.
+
+The process for publishing a blog post is:
+
+1. Create the blog post file
+1. Create a pull request
+1. Confirm via the preview deployment site that everything is looking good (the `deploy/netlify — Deploy preview ready!` link in the checks section)
+1. Get your pull request approved n(preferably by someone on the marketing team)
```
Comment (nicksnyder) [inline README.md:40]:
```suggestion
1. Get your pull request approved (preferably by someone on the marketing team)
```

You could create a marketing team on GitHub and then assign them as CODEOWNERS so they are automatically added as reviewers for new blog posts.

**Comment by aileenrose** [comment]
> How does this affect Google search authority? (SA)
No effects as long as we have 301 redirects in place.

**Comment by ryan-blunden** [comment]
> How does this affect Google search authority? (SA)
> Let's add redirects for all the new paths. Thanks.

@christinaforney @aileenrose @adamfrankl2015 I should've stated up-front that this **does not** change any of the blog post URLs because that is controlled by the `slug` field which every blog post file has.

**Pull request merged**